samcli/lib/providers/sam_function_provider.py

```diff
@@ -26,7 +26,7 @@ class SamFunctionProvider(FunctionProvider):
     _LAMBDA_LAYER = "AWS::Lambda::LayerVersion"
     _DEFAULT_CODEURI = "."
 
-    def __init__(self, template_dict, parameter_overrides=None):
+    def __init__(self, template_dict, parameter_overrides=None, ignore_code_extraction_warnings=False):
```

Comment:
what is `code extraction warning`?

Reply:
on a processed template. the code uri has changed to become s3 url, on deploy we actually pass in the packaged template. and this prompt kept appearing otherwise.